### PR TITLE
Add proxy support for workstation and appliance VM

### PR DESCRIPTION
**Pull request by nascarsayan**

### CODE REVIEW

**Review by subbartt — APPROVED** [review]

**Comment by nascarsayan** [comment]
@abhi580 Unable to see your comments

File: src/config_avs.jsonc
[the remark this answers is not in the document]
Comment (nascarsayan) [reply src/config_avs.jsonc:1]: Have reverted to json, since parsing JSON with comments works in PowerSHell 7 but not in PowerShell 5.

**Comment by abhi580** [comment]
@nascarsayan : The PR looks good to me. I just wanted to understand if the following Proxy Scenarios are tested/ expected to work:
1. Does this support Proxy Server in AVS/Azure/On-Prem.
2. Does this support reverse and forward proxy scenarios?
3. Does this support SSL and non-SSL proxies.

@msfttusharjuneja : Please add if I am missing any scenarios here

**Comment by nascarsayan** [comment]
> @nascarsayan : The PR looks good to me. I just wanted to understand if the following Proxy Scenarios are tested/ expected to work:
> 
> 1. Does this support Proxy Server in AVS/Azure/On-Prem.
> 2. Does this support reverse and forward proxy scenarios?
> 3. Does this support SSL and non-SSL proxies.
> 
> @msfttusharjuneja : Please add if I am missing any scenarios here

@abhi580 , Have tested the bash and powershell script in both SSL proxy, non-ssl proxy.Have made some improvements to the bash script.
- The forward / reverse proxies are different scenarios. Reverse proxy is what we typically do when we host some server in an internal network and want to expose it outside to the public. That is not applicable here.
- Proxy server can be anywhere, that depends on customer's infra. Our script will support any kind of http / https proxy server.

**Review by abhi580 — APPROVED** [review]

**Pull request merged**